### PR TITLE
CSS Guide Page

### PR DESCRIPTION
**Pull request by zacjones93**
Loom walk through: https://www.loom.com/share/2d09fc12574b432fb039ac0fbf024500

This converts a Learner interview guide into a landing page that we can share more broadly. Content is stored in Sanity so it can be easily updated.

I made the landing page from scratch and recreated simple course and challenge cards

I have yet to do any internal linking to this page so we would need to do that as well. Likely on the Home, CSS, and React pages

woo

(I need to disable an extension for screenshots, ignore the little icon on the right side)
Just enough CSS for Modern App Development  - desktop
Just enough CSS for Modern App Development - mobile

### CODE REVIEW

File: src/pages/guides/just-enough-css-for-modern-app-development.tsx [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+          </div>
+
+          <div className="p-2 mx-auto mt-20 sm:mt-32 rounded-full  w-fit bg-gray-50 dark:bg-opacity-20 dark:text-gray-200 hover:bg-gray-100 relative z-10 cursor-pointer transition-all hover:scale-110">
+            <ArrowDownIcon
+              className="w-5 h-5 text-gray-900 dark:text-white "
+              onClick={() =>
+                scrollRef?.current &&
+                scrollRef?.current.scrollIntoView({behavior: 'smooth'})
+              }
+            />
```
Comment (zacjones93) [inline src/pages/guides/just-enough-css-for-modern-app-development.tsx:243]: scroll to first section with refs

File: src/pages/guides/just-enough-css-for-modern-app-development.tsx [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+          </div>
+        </header>
+        <section className="w-full pb-20">
+          <div className="max-w-screen-lg mx-auto">
+            <div className="flex flex-col-reverse lg:flex-row lg:justify-between mt-20 mb-8 relative">
+              <h2
+                className="lg:mt-20 lg:mb-8 font-medium text-2xl text-center lg:text-left"
+                ref={scrollRef}
```
Comment (zacjones93) [inline src/pages/guides/just-enough-css-for-modern-app-development.tsx:251]: sets the scrollRef

File: src/pages/guides/just-enough-css-for-modern-app-development.tsx [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+              <div className="w-[40ch] mb-8 whitespace-pre-wrap mx-auto sm:mx-0 self-center">
+                {fundamentalSection.description}
+              </div>
+
+              <HorizontalCourseCard course={fundamentalSection.courses[0]} />
+            </div>
+          </div>
+        </section>
+
+        <section className="w-full pt-2 dark:bg-gray-800 bg-gray-50 pb-20">
```
Comment (zacjones93) [inline src/pages/guides/just-enough-css-for-modern-app-development.tsx:277]: alternating background color for sections

File: src/pages/guides/just-enough-css-for-modern-app-development.tsx [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+      byline,
+      path,
+      image,
+      'background': images[label == 'feature-card-background'][0].url,
+      'instructor': collaborators[]->[role == 'instructor'][0]{
+        'name': person->.name
+      },
+    },
+  }
+}`
```
Comment (zacjones93) [inline src/pages/guides/just-enough-css-for-modern-app-development.tsx:385]: groq query

File: src/pages/guides/just-enough-css-for-modern-app-development.tsx [hunk 5]
```diff
@@ -0,0 +1,393 @@
...
+
+export async function getStaticProps(context: any) {
```
Comment (zacjones93) [inline src/pages/guides/just-enough-css-for-modern-app-development.tsx:387]: loaded at build time

File: src/pages/projects/build-modern-layouts-with-css-grid/index.tsx [hunk 6]
```diff
@@ -166,7 +166,7 @@ const landingPage: FunctionComponent<LandingProps> = (props) => {
               <section className="relative mt-16 mb-16 ml-0 mr-0 dark:text-gray-200 md:-ml-28 md:-mr-28">
                 <div className="grid grid-cols-1 gap-5 mt-16 mb-16 lg:grid-cols-2">
                   <div className="relative mb-8 text-center md:mb-0">
-                    <div className="relative z-20 p-4 text-center bg-white shadow-lg dark:bg-gray-800 rounded-xl md:p-8">
+                    <div className="relative z-20 p-4 text-center bg-white shadow-lg dark:bg-gray-800 rounded-xl md:p-8 h-full">
```
Comment (zacjones93) [inline src/pages/projects/build-modern-layouts-with-css-grid/index.tsx:169]: The list of reqs was shorter on the left side which mismatched the two boxes so this fixes

**Review by vojtaholik — APPROVED** [review]
looks great 🔥

**Pull request merged**